### PR TITLE
[ENG-6367] Add Adjustable Pagination Buttons For User Table

### PR DESCRIPTION
**Pull request by Johnetordoff**
-   Ticket: [https://openscience.atlassian.net/browse/ENG-6367]
-   Feature flag: n/a

## Purpose

Add pagination button that allow to set the number of items as well as jumping the first previous, next and last page.

## Summary of Changes

- adds AdjustablePaginator component
- add styling css and template

## Screenshot(s)
Screenshot 2024-10-24 at 3 34 25 PM
Screenshot 2024-10-24 at 3 34 11 PM
Screenshot 2024-10-24 at 3 34 02 PM
Screenshot 2024-10-24 at 3 33 29 PM

### CODE REVIEW

File: translations/en-us.yml [hunk 1]
```diff
@@ -847,22 +849,22 @@ institutions:
             users-by-department: 'Total Users by Department'
             public-vs-private-projects:
                 title: 'Public vs Private Projects'
-                public: 'Public Projects'
```
Comment (Johnetordoff) [inline translations/en-us.yml:850]: For some reason I can not make this go away.

Comment (futa-ikeda) [reply translations/en-us.yml:850]: These seem like it's just removing the trailing space at the end of these, so it's a harmless change

File: lib/osf-components/addon/components/adjustable-paginator/styles.scss
Comment (futa-ikeda) [inline lib/osf-components/addon/components/adjustable-paginator/styles.scss:1]: Would encourage the use of CSS variables in `app/styles/_variables.scss` in this file

File: lib/osf-components/addon/components/paginated-list/layout/template.hbs [hunk 2]
```diff
@@ -50,14 +50,28 @@
...
+                }}
+            </div>
+        {{else}}
+            <div local-class='text-center'>
+                {{simple-paginator
+                    maxPage=this.maxPage
+                    nextPage=(action @next)
+                    previousPage=(action @previous)
+                    curPage=@page
+                }}
```
Comment (futa-ikeda) [inline lib/osf-components/addon/components/paginated-list/layout/template.hbs:72]: I know this is what was there before but can we invoke this component using `<AngleBracket>` invocation like..
```
                <AdjustablePaginator
                    @maxPage={{this.maxPage}}
                    @nextPage={{action @next}}
                    //... the rest of the args
                />
            </div>
        {{else}}
            <div local-class='text-center'>
                <SimplePaginator
                    @maxPage={{this.maxPage}}
                    // ... the rest of the args like this too
                />
```

**Review by futa-ikeda — CHANGES_REQUESTED** [review]
New paginator component looks good. The biggest ask would be to change it to a glimmer component, as we are trying to move away from ember-classic components.

File: lib/osf-components/addon/components/adjustable-paginator/component.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
+import { classNames, tagName } from '@ember-decorators/component';
+import Component from '@ember/component';
```
[the remark this answers is not in the document]
Comment (Johnetordoff) [reply lib/osf-components/addon/components/adjustable-paginator/component.ts:2]: I don't see much point in converting it to glimmer, I tried, but I was just getting little syntax errors here and there and it wasn't worth it.

**Pull request merged**